micro-rdk/src/common/ota.rs

```diff
@@ -0,0 +1,357 @@
+            .fields
+            .get("url")
+            .ok_or(OtaError::ConfigError(
+                "config missing `url` field".to_string(),
+            ))?
+            .kind
+            .as_ref()
+            .ok_or(OtaError::ConfigError(
+                "failed to get inner `Value`".to_string(),
+            ))?
+            .try_into()
+            .map_err(|e: AttributeError| OtaError::ConfigError(e.to_string()))?;
+
+        let url = match kind {
+            Kind::StringValue(s) => s,
+            _ => {
+                return Err(OtaError::ConfigError(format!(
+                    "invalid url value: {:?}",
+                    kind
+                )))
+            }
+        };
+        let connector = OtaConnector::default();
+        Ok(Self {
+            url,
+            connector,
+            exec,
+        })
+    }
+
+    pub(crate) async fn update(&mut self) -> Result<(), OtaError> {
+        let mut uri = self
+            .url
+            .parse::<hyper::Uri>()
+            .map_err(|_| OtaError::ConfigError(format!("invalid url: {}", self.url)))?;
+        if uri.port().is_none() {
+            if uri.scheme_str() != Some("https") {
+                log::error!("no port found and not https");
+            }
+
+            let mut auth = uri
+                .authority()
+                .ok_or(OtaError::Other("no authority present in uri".to_string()))?
+                .to_string();
+            auth.push_str(":443");
+            let mut parts = uri.into_parts();
+            parts.authority = Some(
+                auth.parse()
+                    .map_err(|_| OtaError::Other("failed to parse authority".to_string()))?,
+            );
+            uri = hyper::Uri::from_parts(parts).map_err(|e| OtaError::Other(e.to_string()))?;
+        };
+        let io = self
+            .connector
+            .connect_to(&uri)
+            .map_err(|e| OtaError::Other(e.to_string()))?
+            .await
+            .map_err(|e| OtaError::Other(e.to_string()))?;
+
+        let (mut sender, conn) = {
+            http2::Builder::new(self.exec.clone())
+                .max_frame_size(16_384) // lowest configurable value
+                .timer(H2Timer)
+                .handshake(io)
+                .await
+                .map_err(|e| OtaError::Other(e.to_string()))?
+        };
+
+        let conn = self.exec.spawn(async move {
+            if let Err(err) = conn.await {
+                log::error!("connection failed: {:?}", err);
+            }
+        });
+
+        log::info!("ota connected, beginning download");
+
+        let request = Request::builder()
+            .method("GET")
+            .uri(uri)
+            .body(Empty::<Bytes>::new())
+            .map_err(|e| OtaError::Other(e.to_string()))?;
+        let mut response = sender
+            .send_request(request)
+            .await
+            .map_err(|e| OtaError::Other(e.to_string()))?;
+
+        if response.status() != 200 {
```

Comment:
Please add a TODO here to use `is_success` or similar.

Reply:
[RSDK-9346](https://viam.atlassian.net/browse/RSDK-9346?atlOrigin=eyJpIjoiODBkYzA4MzAzNmUwNDk0Yzg5NGYyZmI0MTM1MGI0MjYiLCJwIjoiaiJ9)